sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -119,4 +133,21 @@ object StringUtils {
       result.toString
```

Comment:
Actually `result` can potentially allocate much more memory than it really needed for `strings`. The `length` summarizes sizes of all appended strings even such strings weren't appended, and the line above: `val result = new java.lang.StringBuilder(length)` could cause OOM which you are trying to avoid in the PR.

Reply:
The `.toString` will allocate a new char array and trim it down to size. So you wouldn't hold on to that extra memory.

But yeah there's a little bit of waste here which should be easy to fix, but not critical.

Reply:
I mean the `java.lang.StringBuilder(length)` will try to allocate big array: http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/AbstractStringBuilder.java#l68 . And it could fail in the allocation independently of the SQL config: `spark.sql.maxPlanLength`.

Reply:
Yes, that's what I said also. But that is not really the cause of the OOM being fixed here - it's the collection of multiple large plans, which this issue by itself wouldn't trigger.

In any case, it's simple to fix.

Reply:
Yes, this was a holdover form an earlier implementation where length was capped at the limit.  WIll fix.